src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,133 @@
+- `section_start`: start of the time interval over which `pfunc` is considered
+- `section_end`: end of the time interval over which `pfunc` is considered
+
+## Description
+The `RateConfig` type allows to specify the functional form of a parametric
+forcing over a time interval. This forcing protocol can then be applied to the parameter
+of a dynamical system using the `RateSystem` constructor, which also allows to modify
+the rate and magnitude of the forcing.
+"""
+mutable struct RateConfig
+    pfunc::Function
+    section_start::Float64
+    section_end::Float64
+end
+
+"""
+    RateSystem
+ 
+"""
+struct RateSystem
+    system
+    forcing
+    pidx
+end
+
+"""
+    RateSystem(sys::ContinuousTimeDynamicalSystem, rc::RateConfig, pidx; kwargs...)
+
+Creates a `RateSystem` type from an autonomous dynamical system `sys` and a time-dependent
+parametric forcing protocol of `RateConfig` type.
+
+## Keyword arguments
+- `forcing_start = RateConfig.section_start`: Time when parameter shift starts (before this, the resulting system will be autonomous)
+- `forcing_length = RateConfig.section_end - RateConfig.section_start`: Time-interval over which RateConfig.pfunc([RateConfig.section_start, RateConfig.section_end]) is spread out (for window_length > RateConfig.section_end - RateConfig.section_start) or squeezed into (for window_length < RateConfig.section_end - RateConfig.section_start)
+- `forcing_scale = 1.0`: Amplitude of the ramping. The ramping is then automatically rescaled 
+- `t0 = 0.0`: Initial time of the resulting non-autonomous system (relevant to later compute trajectories)
+
+## Description
+The returned `RateSystem.system` is is 
+autonomous before `forcing_start`, 
+non-autnonmous from `forcing_start` to `forcing_start+forcing_length` with the parameter shift given by the [`RateConfig`](@def), and again 
+autonomous after `forcing_start+forcing_length`.
+
+## Fields of the resulting `RateSystem` type
+- `system`: Nonautonomous [`CoupledODEs`](@ref) constructed from an autonomous system `sys` and a `RateConfig`
+- `forcing`: Function giving the value of the ramped parameter for every time.
+- `pidx`: Index of the ramped parameter within the parameter container of the autonomous system `sys`.
+
+"""
+function RateSystem(
+    auto_sys::ContinuousTimeDynamicalSystem,
+    rc::RateConfig,
+    pidx;
+    forcing_start=rc.section_start,
+    forcing_length=(rc.section_end - rc.section_start),
+    forcing_scale=1.0,
+    t0=0.0,
+)
+    params = deepcopy(current_parameters(auto_sys))
+    p0 = params[pidx]
+
+    if forcing_start < t0
+        @warn "The forcing starts before the system initial time t0=$(t0)."
+    end
+
+    system = apply_ramping(
+        auto_sys, rc, pidx, p0, params, forcing_start, forcing_length, forcing_scale, t0
+    )
+    forcing = t -> p_modified(t, rc, p0, forcing_start, forcing_length, forcing_scale)
+
+    return RateSystem(system, forcing, pidx)
```

Comment:
I'm missing a function here. Two in fact. We need to API functions to change the forcing level and rate. We should define two functions, `set_forcing_size!(rs, Δp)` and `set_forcing_duration!(rs, Δt)` so that this system can be used downstream to e.g., find the critical forcing rate under constant `Δp`. Or, find the critical parameter increase under constant `Δt`. Etc.

Reply:
Let's also add the `set_forcing_Start!(rs, t0)` for setting the time the forcing starts.